### PR TITLE
build: bump to API33

### LINKED ISSUE #14120: [BUG]: Media is not synced until media permission is given in full builds
### Checked for duplicates?

- [X] This issue is not a duplicate

### What are the steps to reproduce this bug?

Try syncing with media sync enabled

### Expected behaviour

Media is synced

### Actual behaviour

Media isn’t synced

### Debug info

```text
2.16.0beta5, Android 13, full version
```

### (Optional) Anything else you want to share?

Only started syncing after I tapped check media and gave the media access permission

### Research

- [X] I am reporting a bug specific to AnkiDroid (Android app)
- [X] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [X] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### LINKED ISSUE #14129: Update to targetSdkVersion 33 - ideally in 2.16, deadline Oct 31 2023
_[Starting on August 31, 2023:
    App updates must target Android 13 or higher and adjust for behavioral changes in Android 13; 
except for Wear OS apps, which must target Android 11.](https://developer.android.com/google/play/requirements/target-sdk)_

Issue to track things that need to be changed when targetSdkVersion is upgraded to 33

### PR DESCRIPTION
**Pull request by mikehardy**
## Purpose / Description

Use API33 as our target, to comply with Play Store guidelines

## Fixes
* Fixes #14129 
* Fixes #14120

## Approach
- bumps the targetSdkVersion
- temporarily clamps robolectric to API32 until they fix an upstream issue (as noted in separate commit)

## How Has This Been Tested?

I tested these things, as indicated in related issue:

- system follows theme with light and dark in system, and app-specific them in AnkiDroid prefs with light and dark both checked
  - changelog loading (followed theme)
  - Anki manual, Github FAQ (followed theme)
  - AnkiDroid manual (did not follow theme, because there is no theming on AnkiDroid.org - this is actually correct app behavior...)
- acquisition of audio, image, video without specifying new API33 granular permissions, both when directly creating them during card creation and when using the picker to choose existing ones

Everything worked for me

## Learning (optional, can help others)

Sometimes it is easy !?! 

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/test/resources/robolectric.properties [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Android 13 triggers robolectric issue: https://github.com/robolectric/robolectric/issues/8215
+# Temporarily lock to sdk31 until that issue is resolved, then remove this file so
```
Comment (david-allison) [inline AnkiDroid/src/test/resources/robolectric.properties:2]:
```suggestion
# Temporarily lock to sdk32 until that issue is resolved, then remove this file so
```

Comment (mikehardy) [reply AnkiDroid/src/test/resources/robolectric.properties:2]: Caught in the cross-fire while you were doing 32, I had started from 31 then pulled yours and rebased on to it, then rebased it out 😆 . Nice to be moving quickly again! It's so refreshing.

**Review by david-allison — COMMENTED** [review]
* ⚠️ Check Media fails/hangs after requesting permissions (`full` build)
  * This may not be a regression caused by this PR
  * This may be fixed by #14130
  * Starts to work after restarting the app

I haven't tested and would like to see a manual test of an app using `CardContentProvider:insertMediaFile` - this API wasn't unit tested to the level that other external-facing APIs were

Other than that, looks good!

====

On the Check Media front: 

* We should revisit 
  * https://github.com/ankidroid/Anki-Android/pull/10926
  * https://github.com/ankidroid/Anki-Android/issues/10892

I strongly suspect now we've moved to 'full', this is no longer necessary and we can remove the permission check

**Comment by mikehardy** [comment]
We have some time and I was hoping to get #14130 in to 2.16, perhaps we focus first on that one (or test with this PR rebased on to it...) to see if that resolves it or alters behavior?

Thanks for the testing and deep-links to other items...

**Comment by BrayanDSO** [comment]
As I noted there, #14130 will need changes to separe the `Photos and Videos` from the `Audio and music` permissions if `targetSdkVersion = 33`. I think I have a branch with the change, so I can commit it there. 

And about the order, if we are going to do upgrade the targetSdk in 2.16, I'd merge this first

**Comment by david-allison** [comment]
Those permission requests were a hack due to what I believe was a bug in android permissions:

* `WRITE_EXTERNAL_STORAGE` was granted without `READ_EXTERNAL_STORAGE`

This occurred in Android 11, and potentially other OS versions after an upgrade to AnkiDroid.

On Android 13 `READ_EXTERNAL_STORAGE` has been removed, it doesn't seem feasible to keep this complexity around an Android bug. Especially as this bug should no longer exist exist (if `READ` has been removed, how can it be denied?) We almost certainly can't grant it any more, and the new permissions don't map to `READ` in the same way.

**Proposal:** 

Change:

https://github.com/ankidroid/Anki-Android/blob/8c00aa836b5322c78abef7d5982514180c11d98b/AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt#L257

to 

```kotlin
if (Build.VERSION.SDK_INT < Build.VERSION_CODES.TIRAMISU) arrayOf(Manifest.permission.READ_EXTERNAL_STORAGE, Manifest.permission.WRITE_EXTERNAL_STORAGE) else emptyArray(),
````

along with a comment.

* Get this PR in
* Remove the requests for `READ_MEDIA_IMAGES` and `READ_MEDIA_VIDEO` from #14130

**Comment by BrayanDSO** [comment]
Remember to check if media is being synced as well without the photos, videos and audios permissions (#14120). And I can't tweak #14130 much for the moment, so I'm happy with anyone pushing changes there

**Comment by david-allison** [comment]
@BrayanDSO Can you reproduce the above issue after `targetSdkVersion 33` is set?

If so:
* Are both of the `READ_` permissions required to fix the issue
* Or does the legacy `READ_EXTERNAL_STORAGE` permission still need to be granted?

**Comment by BrayanDSO** [comment]
> Can you reproduce the above issue after targetSdkVersion 33 is set?

good thing. It solves it

**Comment by david-allison** [comment]
Awaiting #14261, then:



<details>
<summary>Proposed commit notes & patch</summary>


```
fix: on Android 13 don't request EXTERNAL_STORAGE

We requested: [`WRITE_EXTERNAL_STORAGE`, `READ_EXTERNAL_STORAGE`]
due to a bug in Android where `WRITE_EXTERNAL_STORAGE` could be
  granted without `READ_EXTERNAL_STORAGE`.

This meant that a user could not access the /AnkiDroid folder
(but it did work from the WebView). This blocked 'Check Media'
and 'Media Sync'.

Re-requesting `READ_EXTERNAL_STORAGE` fixed the permission issue.

Now that we are on `targetSdkVersion 33`, this bug can no longer be
reproduced on Android 13. [See discussion]

As Android 13 also replaces `READ_EXTERNAL_STORAGE` with:
[`READ_MEDIA_IMAGES`, `READ_MEDIA_VIDEO`, `READ_MEDIA_AUDIO`],
these permissions no longer relate, and it makes sense to stop
applying this workaround in case it's still relevant
(as the permissions no longer match the intent)

So: no longer check for these permissions on Android 13

Original bug: 10892

Fixes 14120
Discussion: 14259
```

```patch
Subject: [PATCH] fix: on Android 13 don't request EXTERNAL_STORAGE
---
Index: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt b/AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt
--- a/AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt	(revision 9a345490ed02f3d8fe7cb3e0f1b92750cd6688aa)
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt	(revision 6b8d62c3f2ef77a1f91f73c45442c69158a80034)
@@ -254,7 +254,7 @@
     // used for check media
     private val checkMediaStoragePermissionCheck = PermissionManager.register(
         this,
-        arrayOf(Manifest.permission.READ_EXTERNAL_STORAGE, Manifest.permission.WRITE_EXTERNAL_STORAGE),
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.TIRAMISU) arrayOf(Manifest.permission.READ_EXTERNAL_STORAGE, Manifest.permission.WRITE_EXTERNAL_STORAGE) else emptyArray(),
         useCallbackIfActivityRecreated = true,
         callback = callbackHandlingStoragePermissionsCheckForCheckMedia(WeakReference(this))
     )
```

</details>

**Comment by BrayanDSO** [comment]
https://github.com/robolectric/robolectric/issues/8215 was solved

**Review by david-allison — APPROVED** [review]
rebased + added commit + force pushed

Good to go now

**Comment by david-allison** [comment]
From Mike

> The robolectric thing should be peeled to a separate issue - the underlying upstream thing is fixed but I'm fine using API32 for robolectric until upstream fix is released in non-SNAPSHOT form. So the issue is just to remove the robolectric.properties sdk=32 clamp once robolectric releases version n+1

**Review by mikehardy — COMMENTED** [review]
Nice! Would you believe we actually *made* the deadline, in the sense we could release this thing no problem before Aug 31 Play Store deadline

I requested more time on the deadline so no rush, but would be nice to put this in the beta pipeline Monday Aug 28 and release it anyway in case it does affect play store visibility in any way

**Pull request merged**